<launch>

  <!-- Required arguments -->
  <arg name="config"/>
  <arg name="out_dir"/>
  <arg name="data_dir"/>
  <arg name="scan_id"/>
  
  <arg name="namespace" default="data"/> 

  <!-- ScanNet data player -->
  <node pkg="active_panoptic_mapping_utils" type="scannet_data_player.py" name="scannetv2_data_player" output="screen">
    <param name="scans_dir_path" value="$(arg data_dir)" /> 
    <param name="scan_id" value="$(arg scan_id)" />
    <param name="play_rate" value="0.1" /> 
    <param name="global_frame_name" value="world" /> 
    <param name="sensor_frame_name" value="depth_camera" /> 
    <param name="wait" value="true" /> 
    <param name="use_uncertainty" value="true" />
    <param name="image_width" value="320" />
    <param name="image_height" value="240"/>
    <param name="panoptic_seg_dir" value="mask2former_swin_tiny"/>
            
    <remap from="~color_image" to="$(arg namespace)/color_image" />  
    <remap from="~depth_image" to="$(arg namespace)/depth_image" />  
    <remap from="~id_image" to="$(arg namespace)/segmentation_image" />  
    <remap from="~uncertainty_image" to="$(arg namespace)/uncertainty_image" />
    <remap from="~labels" to="$(arg namespace)/segmentation_labels" />  
    <remap from="~pose" to="$(arg namespace)/pose" />
  </node>

  <node name="panoptic_mapper" pkg="panoptic_mapping_ros" type="panoptic_mapper_node" output="screen" required="true">
    <!-- Config -->
    <rosparam file="$(find active_panoptic_mapping_utils)/config/mapper/single_tsdf_base.yaml"/>
    <rosparam file="$(arg config)" command="load"/>
    <param name="shutdown_when_finished" value="true"/>

    <!-- Override experiment specific parameters -->
    <param name="save_map_path_when_finished" value="$(arg out_dir)/final"/>
    <param name="labels/file_name" value="$(find active_panoptic_mapping_utils)/config/mapper/nyu40_detectron_labels.csv"/>
    <param name="data_writer/output_directory" value="$(arg out_dir)"/>
     
    <!-- Input -->
    <remap from="color_image_in" to="$(arg namespace)/color_image"/>
    <remap from="depth_image_in" to="$(arg namespace)/depth_image"/>
    <remap from="segmentation_image_in" to="$(arg namespace)/segmentation_image"/>
    <remap from="labels_in" to="$(arg namespace)/segmentation_labels"/>
    <remap from="uncertainty_image_in" to="$(arg namespace)/uncertainty_image"/>
  </node>  

</launch>